src/main/java/weizberg/citibike/ui/CitibikeController.java

```diff
@@ -0,0 +1,71 @@
+package weizberg.citibike.ui;
+
+import org.jxmapviewer.viewer.DefaultWaypoint;
+import org.jxmapviewer.viewer.Waypoint;
+import weizberg.citibike.aws.CitibikeRequest;
+import weizberg.citibike.aws.CitibikeResponse;
+import weizberg.citibike.aws.CoordinateLocation;
+import weizberg.citibike.lambda.LambdaServiceFactory;
+
+import java.io.IOException;
+
+
+public class CitibikeController {
+
+    CitibikeRequest request;
+    LambdaServiceFactory serviceFactory;
+    CitibikeResponse response;
+
+    public CitibikeController() {
+        request = new CitibikeRequest();
+        serviceFactory = new LambdaServiceFactory();
+    }
+
+    public void retrieveBikeInformation(String strFromLocation, String strToLocation) throws IOException {
+        addFromLocation(strFromLocation);
+        addToLocation(strToLocation);
+        sendRequest();
+    }
+
+    public Waypoint startLocation() {
+        return new DefaultWaypoint(request.getFrom().getLat(), request.getFrom().getLon());
+    }
+
+    public Waypoint endLocation() {
+        return new DefaultWaypoint(request.getTo().getLat(), request.getTo().getLon());
+    }
+
+
+    public Waypoint bikePickUpLocation() {
+        return new DefaultWaypoint(response.getStart().getLat(), response.getStart().getLon());
+    }
+
+    public Waypoint bikeDropOffLocation() {
+        return new DefaultWaypoint(response.getEnd().getLat(), response.getEnd().getLon());
+    }
+
+
+    public void sendRequest() throws IOException {
+        response = serviceFactory.callLambda(request);
```

Comment:
This is a synchronous call.

You can't do this (or use blocking get) in your application UI code